server/src/domain/person/person.dto.ts

```diff
@@ -16,24 +18,55 @@ export class PersonUpdateDto {
   @IsOptional()
   @IsString()
   featureFaceAssetId?: string;
+
+  /**
+   * Person visibility
+   */
+  @IsOptional()
+  @IsBoolean()
+  isHidden?: boolean;
+}
+
+export class PersonCountResponseDto {
+  @ApiProperty({ type: 'integer' })
+  hidden!: number;
+  @ApiProperty({ type: 'integer' })
+  visible!: number;
+  @ApiProperty({ type: 'integer' })
+  total!: number;
 }
 
 export class MergePersonDto {
   @ValidateUUID({ each: true })
   ids!: string[];
 }
 
+export class PersonSearchDto {
+  @IsBoolean()
+  @Transform(toBoolean)
+  withHidden?: boolean = false;
+}
+
 export class PersonResponseDto {
   id!: string;
   name!: string;
   thumbnailPath!: string;
+  isHidden!: boolean;
+}
+
+export class PeopleResponseDto {
+  total!: number;
+  visible!: number;
+  @IsArray()
```

Comment:
Response dtos don't get validated by class validator, so this decorator isn't being applied and can be removed.